src/service-worker/core.js

```diff
@@ -0,0 +1,299 @@
+/**
+ * Copyright 2016 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import '../../third_party/babel/custom-babel-helpers';
+import indexedDBP from "../../third_party/indexed-db-as-promised/index"
+
+/**
+ * The SW's current version. We conveniently use JS Date timestamps as version
+ * numbers, so we use this to determine how old a given versioned file is
+ * compared to the latest version (this version).
+ * @const
+ */
+const VERSION = '$internalRuntimeVersion$';
+
+/**
+ * The release date of the latest version.
+ * @const
+ */
+const RELEASE_DATE = versionToDate(VERSION);
+
+/**
+ * Returns the release date of a given AMP release version.
+ *
+ * @param {string} ampVersion
+ * @return {!Date}
+ */
+function versionToDate(ampVersion) {
+  return new Date(parseInt(ampVersion, 10));
+}
+
+/**
+ * Returns the version of a given versioned JS file.
+ *
+ * @param {string} url
+ * @return {string}
+ */
+function ampVersion(url) {
+  const matches = /rtv\/(\d+)/.exec(url);
```

Comment:
Hmm, I think this is bad unless I'm missing something. Prod isn't always `01`. When we flip experiments, this number can get higher.

One thing: We should not do arithmetic with the version number. The only thing that counts is the age of the cached version. We might not push for many weeks and don't want to stop ever trusting the cache contents if the version happens to have been created a few weeks ago.